api/tests/test_users_api.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from django.urls import reverse
+
+
+@pytest.mark.django_db
+def test_normal_user_can_view_only_self(user_api_client, user_2):
+    response = user_api_client.get(
+        reverse("user-list"),
+        format="json",
+    )
+    assert response.status_code == 200
+    assert user_2.id not in map(lambda x: x["id"], response.data)
+
+
+@pytest.mark.django_db
+def test_user_can_view_current_user(user_api_client, user):
+    response = user_api_client.get(
+        reverse("user-current"),
+        format="json",
+    )
+
+    assert response.status_code == 200
+    assert response.data["id"] == user.id
+
+
+def test_general_admin_can_view_all_users(
+    general_admin_api_client, general_admin, user, user_2
+):
+    response = general_admin_api_client.get(
+        reverse("user-list"),
+        format="json",
+    )
+
+    assert response.status_code == 200
+    user_id_list = list(map(lambda x: x["id"], response.data))
+
+    assert user.id in user_id_list
```

Comment:
Could use assertpy assertions for more fluent assertions, like 

assert_that(user_id_list).contains_only(user.id, user_2.id, general_admin.id)

But we can  refactor later if you're running out of hours.

Reply:
Changed in a few places